CONTRIBUTING.md

```diff
@@ -45,11 +45,39 @@ To start developing on AvalancheGo, you'll need a few things installed.
 ./scripts/protobuf_codegen.sh
 ```
 
-- To add or remove an interface that needs a corresponding mock generated, add it to the mock file [here](./scripts/mocks.mockgen.txt). You can regenerate the mocks by running the following script.
+#### Autogenerated mocks
 
-```sh
-./scripts/mock.gen.sh
-```
+Mocks are auto-generated using [mockgen](https://pkg.go.dev/go.uber.org/mock/mockgen) and `//go:generate` commands in the code.
+
+- To **re-generate all mocks**, use the command below from the root of the project:
+
+    ```sh
+    go generate -run "go.uber.org/mock/mockgen" ./...
+    ```
+
+- To **add** an interface that needs a corresponding mock generated:
+  - if the file `mocks_generate_test.go` exists in the package where the interface is located, either:
+    - modify its `//go:generate go run go.uber.org/mock/mockgen@v0.5` to generate a mock for your interface (preferred); or
+    - add another `//go:generate go run go.uber.org/mock/mockgen@v0.5` to generate a mock for your interface according to specific mock generation settings
+  - if the file `mocks_generate_test.go` does not exist in the package where the interface is located, create it with content (adapt as needed):
+
+    ```go
+    // Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+    // See the file LICENSE for licensing terms.
+
+    package mypackage
+
+    //go:generate go run go.uber.org/mock/mockgen@v0.5 -package=${GOPACKAGE} -destination=mocks_test.go . YourInterface
+    ```
+
+    Notes:
+    1. Ideally generate all mocks to `mocks_test.go` for the package you need to use the mocks for and do not export mocks to other packages. This reduces package dependencies, reduces production code pollution and forces to have locally defined narrow interfaces.
```

Comment:
This is really the best for the reasons enumerated, at the cost of having some generated mock code duplication here and there. Note I'm not advising to migrate the current mock file/exported setup, but it's best to use this approach for new code though.